### PR TITLE
Specify how to enable ssl hostname verification for python client

### PR DESCRIPTION
**Pull request by AurelienSylvan**
### Description
Python client documentation does not specify how to properly enable ssl hostname verification. The trick is that the `False` value when disabled needs to become a string when enabled, which is confusing.

### Issues Resolved
None

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
@saimedhi - Can you please review/sign off on this for technical accuracy? Thank you.

**Comment by saimedhi** [comment]
Hello @AurelienSylvan,

May I know if you've tested using ssl_assert_hostname as a comma-separated list of hosts to enable hostname verification?

From what I've gathered in the urllib3 [documentation](https://urllib3.readthedocs.io/en/stable/advanced-usage.html#verifying-tls-against-a-different-host):

- We typically use ssl_assert_hostname when the server presents a certificate different from the hostname or the SNI hostname. [here](https://urllib3.readthedocs.io/en/stable/advanced-usage.html#verifying-tls-against-a-different-host)

- It can be set to values like False, None, or an actual hostname. [here](https://github.com/urllib3/urllib3/issues/2126#issuecomment-783942224)

- By default, assert_hostname is set to server_hostname. [here](https://github.com/urllib3/urllib3/pull/2169#issue-824446694)

- Documentation on using different hosts with distinct parameters can be found [here](https://github.com/opensearch-project/opensearch-py/blob/main/opensearchpy/client/__init__.py#L102).

- As far as I understand, the term "comma-separated list of hosts to enable hostname verification" can be misleading when dealing with multiple hosts.

Thanks for your contribution! :)

**Comment by AurelienSylvan** [comment]
@saimedhi
 You're right. I've just tested with a list, and verification fails.
I don't remember why I thought it should be a list, maybe I got confused by the error message `AttributeError('bool' object has no attribute 'strip')`.

Anyway, i'll correct that.
Thank you for your review.

**Review by saimedhi — CHANGES_REQUESTED** [review]
Hello @AurelienSylvan,

I know it's challenging to cover everything in just one line. I believe it would be beneficial to include detailed explanations, such as when to use it, its default value, and examples, in the opensearch-py guide [here](https://github.com/opensearch-project/opensearch-py/blob/main/guides/ssl.md). We can then reference this documentation as needed. What are your thoughts on this approach?

**Comment by AurelienSylvan** [comment]
Hello @saimedhi 
Indeed, we  can be more verbose. I've just added more details about ssl hostname verification, and removed the comments. WDYT ?

I don't get your point concerning the opensearch-py guide. Are you saying we should move this documentation into the opensearch-py repo ? Or add a reference somewhere ?

**Comment by saimedhi** [comment]
> I don't get your point concerning the opensearch-py guide. Are you saying we should move this documentation into the opensearch-py repo ? Or add a reference somewhere ?

Thanks for the quick response, @AurelienSylvan! I've submitted a [PR](https://github.com/opensearch-project/opensearch-py/pull/694) in opensearch-py for the ssl_assert_hostname guide. You're welcome to check it out and leave any comments or suggestions. Feel free to use it as a reference for your own PR.

File: _clients/python-low-level.md
Comment (saimedhi) [inline _clients/python-low-level.md:112]: To enable hostname verification, **set verify_certs to True**. 

Please review this [PR](https://github.com/opensearch-project/opensearch-py/pull/694) at your convenience. You may also wait until it's merged to avoid potential revisions and include some content here.

File: _clients/python-low-level.md
Comment (saimedhi) [inline _clients/python-low-level.md:119]: I believe this case isn't necessary. We can just add it as a comment in the previous case, as seen in this [PR](https://github.com/opensearch-project/opensearch-py/pull/694/files#diff-c30be5e16358678f3c4a1a369fb517f6869a129560d7b04fb9144b64813a1d01R57). Feel free to share your thoughts on the PR if you have any other ideas. Thank you :)

**Comment by saimedhi** [comment]
Hello @AurelienSylvan! I'm pleased to inform you that the pertinent [guide](https://github.com/opensearch-project/opensearch-py/blob/main/guides/ssl.md) has now been added to opensearch-py. Can we please close the PR? Thank you sincerely for your contributions!

**Comment by AurelienSylvan** [comment]
Yep ! Let's close it. Thank you.

**Comment by AurelienSylvan** [comment]
Closing this issue as [documentation](https://github.com/opensearch-project/opensearch-py/blob/main/guides/ssl.md) has been added to the opensearch-py library about handling ssl hostname verification.

**Pull request closed without merging**
